client/src/legacy/CMSMain.EditForm.js

```diff
@@ -476,4 +477,50 @@ $.entwine('ss', function($){
 	if ($('.cms-edit-form.CMSPageSettingsController input[name="ParentType"]:checked').attr('id') == 'Form_EditForm_ParentType_root') {
 		$('.cms-edit-form.CMSPageSettingsController #Form_EditForm_ParentID_Holder').hide(); //quick hide on first run
 	}
+
+  var confirmed = false;
+  /**
+   * Warn the user not to remove the homepage
+   */
+  $('.cms-edit-form .btn-toolbar #Form_EditForm_action_unpublish.homepage-warning,' +
+	'.cms-edit-form .btn-toolbar #Form_EditForm_action_archive.homepage-warning,' +
+	'#Form_EditForm_URLSegment_Holder.homepage-warning .btn.update').entwine({
+    onclick: async function (e) {
+      if (confirmed) {
+        return this._super(e);
+      }
+      e.stopPropagation();
+
+      var message = i18n.sprintf(
+        i18n._t(
+          'CMS.RemoveHomePageWarningMessage',
+          'Warning: This page has the URL of %s. ' +
+          'This is required by your website to  identify the home page. ' +
+          'Without this, visitors will not be able to view your home page.'
+        ),
+        "<em>home</em>"
```

Comment:
You can use `i18n.inject()` to inject placeholders, instead of sprintf. I'm not sure why you need it here though, since the injection is hardcoded as home